src/models/newJobHandler.ts

```diff
@@ -31,6 +33,79 @@ export class NewJobHandler implements IJobHandler<IngestionJobParams> {
     }
   }
 
+  private async getOrCreatePolygonPartsEntity(
+    job: IJobResponse<IngestionJobParams, unknown>,
+    validatedRequestBody: PolygonPartsPayload
+  ): Promise<PolygonPartsEntityNameObject> {
+    const existingEntityName = job.parameters.additionalParams.polygonPartsEntityName as string | undefined;
+
+    if (existingEntityName !== undefined) {
+      return this.handleRetryScenario(existingEntityName, validatedRequestBody);
+    }
+
+    return this.handleFirstAttempt(validatedRequestBody);
+  }
+
+  private async handleRetryScenario(existingEntityName: string, validatedRequestBody: PolygonPartsPayload): Promise<PolygonPartsEntityNameObject> {
+    try {
+      const existingEntity = await this.checkExistingPolygonPartsEntity(validatedRequestBody);
+      return existingEntity ?? (await this.createPolygonParts(validatedRequestBody));
+    } catch (error) {
+      if (error instanceof ConflictError) {
+        return this.handleConflictInRetry(existingEntityName);
+      }
+      throw error;
+    }
+  }
+
+  private async handleFirstAttempt(validatedRequestBody: PolygonPartsPayload): Promise<PolygonPartsEntityNameObject> {
+    try {
+      return await this.createPolygonParts(validatedRequestBody);
+    } catch (error) {
+      if (error instanceof ConflictError) {
```

Comment:
if you catched an conflict error why did you throw error with polygonPartsEntityName is missing